ospfd/ospf_vty.c

```diff
@@ -8859,6 +8862,7 @@ DEFUN (no_ip_ospf_area,
 
 	if (ospf) {
 		ospf_interface_area_unset(ospf, ifp);
+		ospf_area_check_free(ospf, area_id);
```

Comment:
Also could you, please, move this call after `ospf->if_ospf_cli_count--`, because unsettling the area and decrementing the counter are connected actions and it is more clear when they are executed together.